src/components/modebar/manage.js

```diff
@@ -99,10 +99,19 @@ function getButtonGroups(fullLayout, buttonsToRemove, buttonsToAdd) {
 
     var hasCartesian = fullLayout._hasCartesian,
         hasGL2D = fullLayout._hasGL2D,
-        allAxesFixed = areAllAxesFixed(fullLayout);
+        allAxesFixed = areAllAxesFixed(fullLayout),
+        dragModeGroup = [];
+
+    if((hasCartesian || hasGL2D) && !allAxesFixed) {
+        dragModeGroup = ['zoom2d', 'pan2d'];
+    }
+    if(hasCartesian) {
+        dragModeGroup.push('select2d');
+        dragModeGroup.push('lasso2d');
```

Comment:
@alexcjohnson (cc @chriddyp @jackparmer )

Should we show the select-box and lasso by default on all cartesian graphs and leave it up to plotly.js users or widget devs to add them using `config.modeBarButtons` ?

It seems strange to add two buttons that are useless for all but one trace type + mode combination on all cartesian graphs.

Alternatively, we could make `getButtonsGroups` in the mode bar manager look into fullData for scatter with marker traces and only show the buttons when present.